src/SQLStore/PropertyTableInfoFetcher.php

```diff
@@ -0,0 +1,219 @@
+	}
+
+	/**
+	 * Find the id of a property table that is suitable for storing values of
+	 * the given type. The type is specified by an SMW type id such as '_wpg'.
+	 * An empty string is returned if no matching table could be found.
+	 *
+	 * @since 1.8
+	 *
+	 * @param string $dataTypeTypeId
+	 *
+	 * @return string
+	 */
+	public function findTableIdForDataTypeTypeId( $dataTypeTypeId ) {
+		return $this->findTableIdForDataItemTypeId(
+			DataTypeRegistry::getInstance()->getDataItemId( $dataTypeTypeId )
+		);
+	}
+
+	/**
+	 * Find the id of a property table that is normally used to store
+	 * data items of the given type. The empty string is returned if
+	 * no such table exists.
+	 *
+	 * @since 1.8
+	 *
+	 * @param integer $dataItemId
+	 *
+	 * @return string
+	 */
+	public function findTableIdForDataItemTypeId( $dataItemId ) {
+
+		if ( array_key_exists( $dataItemId, $this->defaultDiTypeTableIdMap ) ) {
+			return $this->defaultDiTypeTableIdMap[$dataItemId];
+		}
+
+		return '';
+	}
+
+	/**
+	 * Retrieve the id of the property table that is to be used for storing
+	 * values for the given property object.
+	 *
+	 * @since 1.8
+	 *
+	 * @param DIProperty $property
+	 *
+	 * @return string
+	 */
+	public function findTableIdForProperty( DIProperty $property ) {
+
+		if ( $this->fixedPropertyTableIds === null ) {
+			$this->getPropertyTableDefinitions();
+		}
+
+		$propertyKey = $property->getKey();
+
+		if ( array_key_exists( $propertyKey, $this->fixedPropertyTableIds ) ) {
+			return $this->fixedPropertyTableIds[$propertyKey];
+		}
+
+		return $this->findTableIdForDataTypeTypeId( $property->findPropertyTypeID() );
+	}
+
+	/**
+	 * Return the array of predefined property table declarations, initialising
+	 * it if necessary. The result is an array of SMWSQLStore3Table objects
+	 * indexed by table ids.
+	 *
+	 * It is ensured that the keys of the returned array agree with the name of
+	 * the table that they refer to.
+	 *
+	 * @since 2.2
+	 *
+	 * @return TableDefinition[]
+	 */
+	public function getPropertyTableDefinitions() {
+
+		if ( $this->propertyTableDefinitions !== null ) {
+			return $this->propertyTableDefinitions;
+		}
+
+		$enabledSpecialProperties = $this->fixedSpecialProperties;
+		$customizableSpecialProperties = array_flip( $this->customizableSpecialProperties );
+
+		foreach ( $this->customSpecialPropertyList as $property ) {
+			if ( isset( $customizableSpecialProperties[$property] ) ) {
+				$enabledSpecialProperties[] = $property;
+			}
+		}
+
+		$definitionBuilder = new PropertyTableDefinitionBuilder(
+			$this->defaultDiTypeTableIdMap,
+			$enabledSpecialProperties,
+			$this->customFixedPropertyList
+		);
+
+		$definitionBuilder->doBuild();
+
+		$this->propertyTableDefinitions = $definitionBuilder->getTableDefinitions();
+		$this->fixedPropertyTableIds = $definitionBuilder->getFixedPropertyTableIds();
+
+		return $this->propertyTableDefinitions;
+	}
+
+	/**
+	 * @since 2.2
+	 */
+	public function clear() {
```

Comment:
`clearCache` is probably better. I'm thinking that having such a method in the interface is not ideal to begin with though. It lets the user know you have a cache. What if you want an implementation that does not have a cache? Not saying this should be fixed, just noting I find it suspicious